research/format.markdown

```diff
@@ -0,0 +1,138 @@
+
+
+
+## Draft 4 to Draft 6:
+
+References: [draft4](https://json-schema.org/draft-04/draft-fge-json-schema-validation-00#rfc.section.7
+) & 
+[draft6](https://json-schema.org/draft-06/draft-wright-json-schema-validation-01#rfc.section.8)
+
+**\+**   **uri-reference** : This attribute applies to string instances. A string instance is valid against this attribute if it is a valid URI Reference (either a URI or a relative-reference), according to [RFC3986].
+
+**\+**   **uri-template** : This attribute applies to string instances. A string instance is valid against this attribute if it is a valid URI Template (of any level), according to [RFC6570].
+
+**\+**    **json-pointer** : This attribute applies to string instances. A string instance is valid against this attribute if it is a valid JSON Pointer, according to [RFC6901].
+
+
+
+## Draft 6 to Draft 7:
+
+References: [draft6](https://json-schema.org/draft-06/draft-wright-json-schema-validation-01#rfc.section.8) & [draft7](https://json-schema.org/draft-07/draft-handrews-json-schema-validation-01#rfc.section.7)
+
+**\+**   **date** : A string instance is valid against this attribute if it is a valid representation according to the "full-date" production.
+
+**\+**   **time** : A string instance is valid against this attribute if it is a valid representation according to the "full-time" production.
+
+Implementations MAY support additional attributes using the other production names defined in that section. If "full-date" or "full-time" are implemented, the corresponding short form ("date" or "time" respectively) MUST be implemented, and MUST behave identically. Implementations SHOULD NOT define extension attributes with any name matching an RFC 3339 production unless it validates according to the rules of that production. [CREF2]
+
+**\+**   **idn-email** : As defined by RFC 6531 [RFC6531]. 
+
+**idn-email ⊃ email**
+
+**\+**   **hostname** : As defined by RFC 1034, section 3.1 [RFC1034], including host names produced using the Punycode algorithm specified in RFC 5891, section 4.4 [RFC5891].
+
+**\+**   **idn-hostname** : As defined by either RFC 1034 as for hostname, or an internationalized hostname as defined by RFC 5890, section 2.3.2.3 [RFC5890].
+
+**idn-hostname ⊃  hostname**
+
+**\+**   **ipv6** : A string instance is valid against this attribute if it is a valid representation of an IPv6 address as defined in RFC 2373, section 2.2 [RFC2373]. 
+
+An IPv6 address as defined in RFC 4291, section 2.2 [RFC4291].
+
+**\+**   **iri**: A string instance is valid against this attribute if it is a valid IRI, according to [RFC3987].
+
+**\+**   **iri-reference**: A string instance is valid against this attribute if it is a valid IRI Reference (either an IRI or a relative-reference), according to [RFC3987].
+**iri ⊃  uri**
```

Comment:
Please avoid non ASCII characters like subset symbol. In prose, you can just say "is a subset of", etc